### ISSUE TITLE
Torro Forms are really hard to style

### ISSUE THREAD

**Issue by Horttcore** [awsmug/torro-forms#503]
## Issue Overview
I've tried to style a multistep torro form and noticed it is really hard to style torroforms 

## Current Behavior
* Can not add a class to the wrapper element
* Insufficient classes at the wrapper elemente

## Possible Solution
* BEM naming convention for all elements
* Add more classes to wrapper

**Comment by mahype** [comment]
That sounds very good! I will take a look for the classes. If you have specific examples or some code I could merge I could speed it a bit up. I would like to use the naming convention but for that I have at first to read the docs of the BEM. I would do that, but I have put this at the end of the todo list at the moment.

Otherwise there is a possibility to add your own classes by the element settings.

![Bildschirmfoto_2019-08-21_um_11_26_04](https://user-images.githubusercontent.com/663697/63420827-ce82bb00-c407-11e9-9632-34aa1e9ca669.jpg)

**Comment by Horttcore** [comment]
@mahype Yeah I've found that, but that class is set to the input, which isn't really helping.
I will look into it, maybe I could open a Pull Request.

Biggest issue for me at the moment is to align form fields like
```
[LastName] [FirstName]
[Street] [Streetnumber]
[ZIP] [City]
[Country]
```

**Comment by mahype** [comment]
> Yeah I've found that, but that class is set to the input, which isn't really helping.
I will look into it, maybe I could open a Pull Request.

Oh that should be fixed. It should surround the element.

**Comment by mahype** [comment]
Maybe this temporary helps you. There is also the fiilter _torro_element_wrap_classes_.

There are a lot of other places where you can fiilter CSS classes. You can find all other CSS filters here:

https://github.com/awsmug/torro-forms/blob/master/src/src/db-objects/elements/element.php#L203

**Comment by Horttcore** [comment]
@mahype Is there a way to get this custom class for the input wrapper?
I think it is not reachable by the `torro_element_wrap_classes` hook.
It is also not possible to `print_r()` the 2nd parameter of `torro_element_wrap_classes` as it produces an infinite loop. So I can't inspect the object. :-/

**Comment by mahype** [comment]
Hi @Horttcore!

For the moment I created a gist for showing how to add CSS classes to the element wrapper. 

https://gist.github.com/mahype/362c0b82dd08c12b54166dccaf1a6b43

But I will move the CSS class which is given in the settings to the element wrapper.

**Comment by mahype** [comment]
I have changed the behaviour now fot the upcoming 1.0.8 version.

I have changed the behaviour now fot the upcoming 1.0.8 version.

**Comment by Horttcore** [comment]
@mahype Thanks for the gist, but I knew that already. What I wanted was to add the custom class for the field also in the wrapper.
`torro-element-wrap--$myClass`

But I can't reach $myClass in the `torro_element_wrap_classes` hook.

**Comment by mahype** [comment]
Sorry, i don't know what you mean. If you want to reach the input classes, you can use the torro_element_input_classes filter. Do you mean that?

**Comment by Horttcore** [comment]
`torro_element_wrap_classes` has to arguments `$classes` and `$element`, where in `$element` is my custom class. I can't `print_r($element)` as it produces an infinite loop.

**Comment by mahype** [comment]
Now I got it. ;) Thats because of the object model. print_r() can't handle recursive infinite data. I have the same issue, but for me works var_dump() or the Xdebug way.

**Comment by Horttcore** [comment]
`<b>Fatal error</b>:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 59854848 bytes) in <b>…</b> on line <b>4339</b><br />`

Mmmhh... still could not print the element.

At least I've found the [line](https://github.com/awsmug/torro-forms/blob/d44ef294b1b629d666414b2d150f66e340bdee41/src/src/db-objects/elements/element.php#L311) where class is added.

But I can't see any possibility to alter any data.
Maybe add an hook [here](https://github.com/awsmug/torro-forms/blob/d44ef294b1b629d666414b2d150f66e340bdee41/src/src/db-objects/elements/element-types/element-type.php#L470), so it is possible that change the element.
